### DISCUSSION TITLE
Denying WebSocket connection in WebSocketEndpoint.on_connect leads to Exception

### DISCUSSION THREAD

**Discussion by dingensundso** [encode/starlette#1555] [Potential Issue] [answered]
I created a WebSocketEndpoint class in which I want to deny the connection in certain conditions.
When the WebSocket is closed in on_connect I receive the following exception:

```
  File "starlette/endpoints.py", line 83, in dispatch
    close_code = int(message.get("code", status.WS_1000_NORMAL_CLOSURE))
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```
So code is None when the server denies the connection instead of non-existant.

https://github.com/encode/starlette/blob/e086fc2da361767b532cf690e5203619bbae98aa/starlette/endpoints.py#L72-L87

Changing line 83 to the following should fix the issue:
```python
close_code = int(message.get("code") or status.WS_1000_NORMAL_CLOSURE)               
````

**Comment by Kludex** [comment, marked as answer]
Created #1560 from this discussion. Thanks for bringing it up. :)
